libnetdata/socket/socket.c

```diff
@@ -105,14 +105,78 @@ char *strdup_client_description(int family, const char *protocol, const char *ip
     return strdupz(buffer);
 }
 
+// --------------------------------------------------------------------------------------------------------------------
+// Validating socket address
+
+/**
+ *  Is Unix socket valid?
+ *
+ *  We are considering valid unix sockets, the sockets that follow all `pathname sockets` specification
+ *  http://man7.org/linux/man-pages/man7/unix.7.html
+ *
+ *  @param path is the file that will be used as socket.
+ *
+ *  @return 1 case it is valid socket path and 0 otherwise
+ */
+int is_unix_socket_valid(const char *path) {
+    struct sockaddr_un tester;
+    size_t length = strlen(path);
+    int ret;
+    if (length >= (sizeof(tester.sun_path)-1))
+        ret =  0;
+    else { //We also check whether there is only ascii text and do not have space
+        ret = 1;
+        while (*path) {
+            if(!isascii(*path) || *path == ' ') {
+                ret = 0 ;
+                break;
+            }
+
+            path++;
+        }
+    }
+
+    return ret;
+}
+
+/**
+ * Is valid tcp hostname
+ *
+ * We are considering valid hostname, addresses that follow the specifications of `hostname` manual
+ * http://man7.org/linux/man-pages/man7/hostname.7.html
+ *
+ * @param host the host that will be used to connect
+ *
+ * @return 1 case it is a valid hostname and 0 otherwise.
+ */
+int is_valid_tcp_hostname(char *host) {
+    int ret = 1;
```

Comment:
Should this be `1` by default? What if `!host || !*host`? Does this function expect host to not be empty?

Reply:
By steps, this function is called from the line 885 and there is already a test for NULL in the line 880, so it does not need to test for Null. I decided do not move the code at 880 inside this function, because the error messages for each case is different.
About the default value to be, it was a good catch. It is not necessary ,so I removed it with the last commit. Thank you!

Reply:
> By steps, this function is called from the line 885 and there is already a test for NULL in the line 880, so it does not need to test for Null. 

I saw it, but this is **dependence on the order excution** which is not safe.  If function expects `host` to not be null it should be in the function description i guess.

Reply:
Now I got your point of view, you are correct, somebody could invert the order and brings a new but for Netdata. I moved the code inside the new function and I did all necessaries adjusts for this.
Thank you again!